### ISSUE TITLE
TypeError: Cannot assign to read only property 'length' of string ''

### ISSUE THREAD

**Issue by razvanphp** [vweevers/tap-completed#2]
Sorry I cannot provide more information, I have no idea why this happens.

```
# cleanup

1..277
# tests 277
# pass  277

# ok

/Volumes/Code/simple-peer/node_modules/minipass/index.js:628
    this.buffer.length = 0
                       ^

TypeError: Cannot assign to read only property 'length' of string ''
    at Parser.destroy (/Volumes/Code/simple-peer/node_modules/minipass/index.js:628:24)
    at Parser.oncomplete (/Volumes/Code/simple-peer/node_modules/tap-completed/index.js:46:7)
    at Parser.emit (events.js:412:35)
    at Parser.emit (/Volumes/Code/simple-peer/node_modules/minipass/index.js:483:23)
    at Parser.emitComplete (/Volumes/Code/simple-peer/node_modules/tap-parser/index.js:584:12)
    at Parser.end (/Volumes/Code/simple-peer/node_modules/tap-parser/index.js:555:10)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! simple-peer@9.11.1 test-browser-local: `airtap --coverage --preset local -- test/*.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the simple-peer@9.11.1 test-browser-local script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/razvan/.npm/_logs/2024-02-11T07_52_20_818Z-debug.log
razvan@Razvans-MBP simple-peer %
```

**Comment by vweevers** [comment]
Known issue but IIRC it shouldn't happen anymore thanks to https://github.com/vweevers/tap-completed/commit/3bb345a0f3b736a4134abd120feb0dc91af03366. Can you check which versions of `tap-completed`, `tap-parser` and `minipass` you have? E.g. by running `npm ls tap-completed`.

**Comment by razvanphp** [comment]
```
razvan@Razvans-MBP simple-peer % npm ls tap-completed
simple-peer@9.11.1 /Volumes/Code/simple-peer
└─┬ airtap@4.0.4
  └── tap-completed@1.0.1

razvan@Razvans-MBP simple-peer % npm ls minipass
simple-peer@9.11.1 /Volumes/Code/simple-peer
├─┬ @koush/wrtc@0.4.9
│ ├─┬ @mapbox/node-pre-gyp@1.0.6
│ │ └─┬ tar@6.1.11
│ │   ├─┬ fs-minipass@2.1.0
│ │   │ └── minipass@3.1.5 deduped
│ │   ├── minipass@3.1.5
│ │   └─┬ minizlib@2.1.2
│ │     └── minipass@3.1.5 deduped
│ └─┬ node-pre-gyp@0.17.0
│   └─┬ tar@4.4.19
│     ├─┬ fs-minipass@1.2.7
│     │ └── minipass@2.9.0
│     ├── minipass@2.9.0
│     └─┬ minizlib@1.3.3
│       └── minipass@2.9.0 deduped
├─┬ airtap-sauce@1.1.2
│ └─┬ airtap-sauce-browsers@0.2.2
│   └─┬ minipass-fetch@1.4.1
│     ├─┬ minipass-sized@1.0.3
│     │ └── minipass@3.3.6 deduped
│     ├── minipass@3.3.6
│     └─┬ minizlib@2.1.2
│       └── minipass@3.3.6 deduped
└─┬ airtap@4.0.4
  └─┬ tap-completed@1.0.1
    ├── minipass@3.1.6
    └─┬ tap-parser@10.1.0
      └── minipass@3.3.6 deduped

razvan@Razvans-MBP simple-peer % npm ls tap-parser
simple-peer@9.11.1 /Volumes/Code/simple-peer
└─┬ airtap@4.0.4
  └─┬ tap-completed@1.0.1
    └── tap-parser@10.1.0

razvan@Razvans-MBP simple-peer %
```

**Comment by vweevers** [comment]
Fixed by #3 which will require a change in airtap. Follow https://github.com/airtap/airtap/issues/334 for progress.
